File: 75-sort-colors/sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {

        int high = nums.size()-1;
        int mid = 0,low = 0;
        while(mid<=high) {
            if(nums[mid] == 0) {
                swap(nums[mid],nums[low]); 
                mid++; low++;
                }
            else if(nums[mid] == 1) 
                mid++;
            else {
                swap(nums[mid],nums[high]);
                high--;
            }
        }

    }
};

File: 75-sort-colors/sort-colors.java
class Solution {
   public void swap(int[] nums, int i, int j) {
    int temp = nums[i];
    nums[i] = nums[j];
    nums[j] = temp;
}

public void sortColors(int[] nums) {
    int low = 0;
    int mid = 0;
    int high = nums.length - 1;
    while (mid <= high) {
        if (nums[mid] == 0) { // Check if the current element is 0
            swap(nums, low, mid); // Swap low and mid
            low++;
            mid++;
        } else if (nums[mid] == 1) { // If it's 1, just move mid forward
            mid++;
        } else { // If it's 2
            swap(nums, mid, high); // Swap mid and high
            high--;
        }
    }
}
}